workspace/server.js

```diff
@@ -1,3 +1,16 @@
 import ultra from "../server.ts";
+import requestHandler from "../src/server/middleware/requestHandler.ts";
 
-ultra();
+const server = ultra();
+
+server.use(async (context, next) => {
+  console.log(`<-- ${context.request.method} ${context.request.url}`);
+  await next();
+  console.log(
+    `--> ${context.request.method} ${context.request.url} ${context.response.status}`,
+  );
+});
+
+server.use(requestHandler());
```

Comment:
Ideally this should be the middleware function at this stage and not a factory
```suggestion
server.use(requestHandler);
```

Reply:
Otherwise (if necessary) `createRequestHandler` will make it clearer.

Reply:
If passing middleware functions directly without factories then there can be no scope outside the request scope. This means that some resources will get re-created for every request. So I think factories are necessary.

Reply:
You can just initialise those outside of function scope at the root instead, the only time that can be an issue though is with circular deps.